src/main/bx/models/mcp/MCPServer.bx

```diff
@@ -0,0 +1,1075 @@
+	 *
+	 * @return This server for chaining
+	 */
+	MCPServer function registerPrompt(
+		required string name,
+		string description = "",
+		array arguments = [],
+		required function handler
+	) {
+		variables.prompts[ arguments.name ] = {
+			"name": arguments.name,
+			"description": arguments.description,
+			"arguments": arguments.arguments,
+			"handler": arguments.handler
+		};
+		return this;
+	}
+
+	/**
+	 * Unregister a prompt by name
+	 *
+	 * @name The name of the prompt to unregister
+	 *
+	 * @return This server for chaining
+	 */
+	MCPServer function unregisterPrompt( required string name ) {
+		variables.prompts.delete( arguments.name );
+		return this;
+	}
+
+	/**
+	 * Check if a prompt is registered
+	 *
+	 * @name The name of the prompt to check
+	 *
+	 * @return Boolean indicating if the prompt exists
+	 */
+	boolean function hasPrompt( required string name ) {
+		return variables.prompts.keyExists( arguments.name );
+	}
+
+	/**
+	 * List all registered prompts in MCP format
+	 *
+	 * @return Array of prompt definitions for MCP protocol
+	 */
+	array function listPrompts() {
+		return variables.prompts.map( ( name, prompt ) => {
+			return {
+				"name": prompt.name,
+				"description": prompt.description,
+				"arguments": prompt.arguments
+			};
+		} ).valueArray();
+	}
+
+	/**
+	 * Get a prompt by calling its handler
+	 *
+	 * @name The name of the prompt
+	 * @arguments The arguments to pass to the prompt handler
+	 *
+	 * @return The prompt messages
+	 */
+	struct function getPrompt( required string name, struct arguments = {} ) {
+		if ( !hasPrompt( arguments.name ) ) {
+			throw( type: "MCPServer.PromptNotFound", message: "Prompt not found: #arguments.name#" );
+		}
+
+		var prompt = variables.prompts[ arguments.name ];
+		var messages = prompt.handler( arguments.arguments );
+
+		// Ensure messages is an array
+		if ( !isArray( messages ) ) {
+			messages = [ messages ];
+		}
+
+		return {
+			"description": prompt.description,
+			"messages": messages.map( msg => {
+				if ( isStruct( msg ) ) {
+					return {
+						"role": msg.role ?: "user",
+						"content": {
+							"type": "text",
+							"text": msg.content ?: ""
+						}
+					};
+				}
+				return {
+					"role": "user",
+					"content": {
+						"type": "text",
+						"text": toString( msg )
+					}
+				};
+			} )
+		};
+	}
+
+	/**
+	 * Clear all registered prompts
+	 *
+	 * @return This server for chaining
+	 */
+	MCPServer function clearPrompts() {
+		variables.prompts = {};
+		return this;
+	}
+
+	/**
+	 * ---------------------------------------------------------------------------------------------------------
+	 * Server Configuration Methods
+	 * ---------------------------------------------------------------------------------------------------------
+	 */
+
+	/**
+	 * Set the server name
+	 *
+	 * @name The server name
+	 *
+	 * @return This server for chaining
+	 */
+	MCPServer function setName( required string name ) {
```

Comment:
All of the setters can be removed, boxlang does this automatically. Both implicit getters and setters for ALL defined properties